### DISCUSSION TITLE
With Windows 11 out, Is a signed ipxe.efi binary still a possibility? Or do we all need our own keys?

### DISCUSSION THREAD

**Discussion by aggieNick02** [ipxe/ipxe#488] [General]
**EDIT: Please see the comment that follows. Windows 11 does not require secure boot be enabled, only that the machine be "Secure Boot capable".**

Now that Windows 11 is out and requires that secure boot be enabled, disabling secure boot to use iPXE to install the latest version of windows or manage boot on a Windows 11 machine is no longer really an option. This leaves those who want to use iPXE with Windows 11 with a couple options:

1. Wait for a signed ipxe.efi binary. ipxe.org has released signed wimboot (https://twitter.com/ipxe/status/1362179198979366919), and stated they're working towards a signed ipxe or snponly (https://twitter.com/nikize0/status/1364615001529540613). Parts of the wimboot work were sponsored by 2Pint Software, who definitely seems to have themselves gotten an ipxe.efi binary signed for their product (https://techcommunity.microsoft.com/t5/hardware-dev-center/ipxe-security-assurance-review/ba-p/1062943, https://techcommunity.microsoft.com/t5/hardware-dev-center/updated-uefi-signing-requirements/ba-p/1062916)
2. Each interested party/company signing ipxe.efi with their own keys and managing those secure boot keys on their machines: https://www.rodsbooks.com/efi-bootloaders/controlling-sb.html

I 100% realize that the community is in no way entitled to a signed binary. It would just be helpful to know the state of things before each of us start down the path for the second option (or perhaps individually undertake submitting a custom-built ipxe.efi to Microsoft for signing). The total development cost of each group having to pursue option 2 seems like it would be orders greater than the cost of option 1, so I wonder if sponsorship or something similar could be helpful.

**Comment by aggieNick02** [comment]
So this isn't the big deal I thought it was. Several tech sites state the requirement for windows 11 is to enable secure boot, but when I went to test, it installed fine when Secure Boot was disabled. Indeed, Microsoft explicitly says only that the PC be secure-boot capable (i.e., booting via any UEFI compliant to the UEFI spec 2.3.1 Errata C or later).
https://docs.microsoft.com/en-us/windows/whats-new/windows-11-requirements (note it says "UEFI, Secure Boot capable")
https://support.microsoft.com/en-us/topic/windows-11-and-secure-boot-a8ff1202-c0d9-42f5-940f-843abef64fad

While it would still be great to know the state of a signed ipxe.efi and what could be done to help, Windows 11 doesn't really change the game for ipxe in the worrying way I thought it did. Sorry for the false alarm.

**Comment by steverweber** [comment]
The https://techcommunity.microsoft.com/t5/hardware-dev-center/ipxe-security-assurance-review/ba-p/1062943 was a good read. I bet @mcb30 enjoyed the compliments :)

I'm just realizing that too many new windows systems are pre installed in secure mode. It is what it is.

**Comment by genekxxx** [comment]
https://knowledge.broadcom.com/external/article/280113/updated-64bit-ipxeefi-ipxe-v1211+-binari.html

**Reply by y2kPArt9** [reply]
> https://knowledge.broadcom.com/external/article/280113/updated-64bit-ipxeefi-ipxe-v1211+-binari.html

Unfortunately it doesn't work on Realtek
